biggus/__init__.py

```diff
@@ -561,6 +563,175 @@ def _normalise_keys(self, keys):
         return keys
 
 
+class ArrayContainer(Array):
```

Comment:
> or refactor other class(es) to make use of it?

That is the idea. ``TransposedArray`` is the first candidate. Many of the rest are also pretty good fits too.